homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -137,6 +144,35 @@ class SynologyDSMFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
 
         return self.async_create_entry(title=host, data=config_data,)
 
+    async def async_step_ssdp(self, discovery_info):
+        """Handle a discovered synology_dsm."""
+        parsed_url = urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION])
+        friendly_name = (
+            discovery_info[ssdp.ATTR_UPNP_FRIENDLY_NAME].split("(", 1)[0].strip()
+        )
+
+        if self._host_already_configured(parsed_url.hostname):
+            return self.async_abort(reason="already_configured")
+
+        # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+        self.context["title_placeholders"] = {
+            CONF_NAME: friendly_name,
```

Comment:
Wait, we _have_ a name here. This means that we should never ask for a name in the config flow.

Reply:
I am actually wondering if we should keep the `name` param @bdraco ...

For the name of sensor, we can use:
```python
self._name = f"Synology DSM {self._api.information.model} {sensor_info[0]}"
```

So it will also addressed @MartinHjelmare comment : https://github.com/home-assistant/core/pull/32704#discussion_r403783143

Reply:
Done.  I've updated the `unique_id` as well

It means you'll have to delete and re-add as the unique ids will change.  This is fine as its only in 0.109 which isn't released.

Reply:
@Quentame The name gets really long with this.
`Synology DSM DS918+ Below Min Remaining Life (sdb)`
How about
`Synology DS918+ Below Min Remaining Life (sdb)`

Reply:
Second is better :wink:

Reply:
Ideally we actually provide the name, but I can't get the api to give it up.  Maybe in a future update. Switching back to `Synology`